modal/parallel_map.py

```diff
@@ -134,13 +134,21 @@ async def pump_inputs():
             logger.debug(
                 f"Pushing {len(items)} inputs to server. Num queued inputs awaiting push is {input_queue.qsize()}."
             )
-            resp = await retry_transient_errors(
-                client.stub.FunctionPutInputs,
-                request,
-                max_retries=None,
-                max_delay=10,
-                additional_status_codes=[Status.RESOURCE_EXHAUSTED],
-            )
+            while True:
+                try:
+                    resp = await retry_transient_errors(
+                        client.stub.FunctionPutInputs,
+                        request,
+                        max_retries=1,
+                        max_delay=101,
+                        additional_status_codes=[Status.RESOURCE_EXHAUSTED],
+                    )
+                    break
+                except GRPCError as err:
+                    if err.status != Status.RESOURCE_EXHAUSTED:
+                        raise err
+                    logger.warning("Unable to push inputs. Check rate of output consumption.")
```

Comment:
Open to suggestions if the wording of the warning message can be improved!

Reply:
For the wording -- maybe something more general like `Warning: map progress is limited. Common bottlenecks include slow iteration over results, or function backlogs.`

Reply:
Thanks - will update!